chpl/chpl-service/src/main/java/gov/healthit/chpl/upload/listing/validation/reviewer/DeveloperReviewer.java

```diff
@@ -0,0 +1,201 @@
+        this.urlValidator = new UrlValidator();
+    }
+
+    public void review(CertifiedProductSearchDetails listing) {
+        Developer developer = listing.getDeveloper();
+        if (developer == null) {
+            listing.getErrorMessages().add(msgUtil.getMessage("listing.missingDeveloper"));
+            return;
+        }
+
+        if (developer.getDeveloperId() != null) {
+            //it's an existing developer - give the user warnings if any required developer data is
+            // 1) missing in the system, or 2) does not match the user-entered data
+            reviewRequiredDeveloperData(listing, WARNING);
+            reviewUserEnteredDataSystemDataMismatch(listing);
+            reviewDeveloperStatusIsActive(listing);
+        } else {
+            String devCode = "";
+            try {
+                devCode = chplProductNumberUtil.getDeveloperCode(listing.getChplProductNumber());
+            } catch (Exception ex) {
+            }
+            if (devCode.equals(DeveloperManager.NEW_DEVELOPER_CODE)) {
+                //it's a new developer - give the user errors if any required data has not been entered
+                reviewRequiredDeveloperData(listing, ERROR);
+            } else {
+                //it's not a new developer by code so it must be an error that no developer was found
+                listing.getErrorMessages().add(msgUtil.getMessage("listing.developer.notFound", listing.getDeveloper().getUserEnteredName()));
+            }
+        }
+    }
+
+    private void reviewUserEnteredDataSystemDataMismatch(CertifiedProductSearchDetails listing) {
+        Developer developer = listing.getDeveloper();
+        if (!StringUtils.isEmpty(developer.getUserEnteredName()) && !developer.getUserEnteredName().equals(developer.getName())) {
+            listing.getWarningMessages().add(msgUtil.getMessage("listing.developer.userAndSystemMismatch", "name", developer.getUserEnteredName(), developer.getName()));
+        }
+        if (!StringUtils.isEmpty(developer.getUserEnteredWebsite()) && !developer.getUserEnteredWebsite().equals(developer.getWebsite())) {
+            listing.getWarningMessages().add(msgUtil.getMessage("listing.developer.userAndSystemMismatch", "website", developer.getUserEnteredWebsite(), developer.getWebsite()));
+        }
+        if (!StringUtils.isEmpty(developer.getUserEnteredSelfDeveloper())) {
+            Boolean userEnteredSelfDeveloper = null;
+            try {
+                userEnteredSelfDeveloper = uploadHandlerUtil.parseBoolean(developer.getUserEnteredSelfDeveloper());
+            } catch (Exception ex) {
+            }
+            if (userEnteredSelfDeveloper != null && !userEnteredSelfDeveloper.equals(developer.getSelfDeveloper())) {
+                listing.getWarningMessages().add(msgUtil.getMessage("listing.developer.userAndSystemMismatch", "self-developer", developer.getUserEnteredSelfDeveloper(), developer.getSelfDeveloper()+""));
```

Comment:
Adding a comment here because I don't see the checkstyle alerts in the main discussion section of the PR